operator/v1/types_machineconfiguration.go

```diff
@@ -56,3 +64,94 @@ type MachineConfigurationList struct {
 	// Items contains the items
 	Items []MachineConfiguration `json:"items"`
 }
+
+type ManagedBootImages struct {
+	// machineManagers is an array of machineManager objects.
+	// The MCO will watch for changes to this list and enroll/un-enroll machine management resources. An entry in this list
+	// consists of the resource type, the API group that the resource belongs to and a selection filter on the resources.
+	// All fields are required.
+	//
+	// Warning: Only one entry per machine management resource type is allowed. If multiple entries target the same type of machine management
+	// resource, all entries associated with that resource type will be ignored. For example, if one entry describes that MAPI machinesets are
+	// disabled for boot image updates and a later entry describes that MAPI machinesets are enabled for boot image updates, both entries
+	// will be ignored.
+	//
+	// +kubebuilder:validation:Required
+	MachineManagers []MachineManager `json:"machineManagers"`
```

Comment:
We should add a maxmium length to this list, I think right now, that might be one, but for that, we need to answer the `partial`

If we were to add a namespace dimension, then the list would obviously be much longer

Reply:
Agree, resolving the caveat will determine the limits on the list